src/groupeddataframe/complextransforms.jl

```diff
@@ -110,7 +118,20 @@ function _combine_rows_with_first!(first::Union{NamedTuple, DataFrameRow},
     # Handle first group
     j = fill_row!(first, outcols, rowstart, colstart, colnames)
     @assert j === nothing # eltype is guaranteed to match
-    # Handle remaining groups
+    # Handle remaining groups (eliminate the argument type-diversity of `first`)
+    _combine_rows_with_first(len, outcols, rowstart, f, gd, incols, colnames, firstmulticol)
+end
+
+function _combine_rows_with_first(len::Int,
```

Comment:
@nalimilan - are you OK with this function name? Maybe `_combine_rows_with_first_filled!` would be better?